test/integration/post.spec.ts

```diff
@@ -0,0 +1,423 @@
+import { HttpStatus, INestApplication } from '@nestjs/common';
+import { DataSource } from 'typeorm';
+import { setDataSource } from 'typeorm-extension';
+
+import {
+  create,
+  getJWTResponse,
+  initApp,
+  initDataSource,
+  mockJwtVerified,
+  getJWTResponseWithFile,
+} from '../helper';
+import { User } from '../../src/entities/user.entity';
+import { Post } from '../../src/entities/post.entity';
+import { Tag } from '../../src/entities/tag.entity';
+import { hash } from '../../src/shared/ultils/bcypt.util';
+
+describe('Post controller', () => {
+  let app: INestApplication;
+  let dataSource: DataSource;
+  let server: any;
+  let user: User;
+  let posts: Partial<Post>[];
+
+  beforeAll(async () => {
+    app = await initApp();
+    server = app.getHttpServer();
+    dataSource = await initDataSource();
+    setDataSource(dataSource);
+    user = await create<User>(User, {
+      password: await hash('123456'),
+    });
+    posts = Array.from({ length: 3 }, (_, index) => ({
+      userId: user.id,
+      title: `title-${index}`,
+      description: `description-${index}`,
+      filePath: `filePath-${index}`,
+    }));
+    await Promise.all(posts.map((post: Post, index: number) => {
+      const created = new Date();
+      created.setMinutes(created.getMinutes() - index);
+      return create(Post, { ...post, created });
+    }));
+  });
+
+  afterAll(async () => {
+    server.close();
+    jest.clearAllMocks();
+
+    await dataSource.getRepository(Post).delete({});
+    await dataSource.getRepository(User).delete({});
+    await app.close();
+  });
+
+  describe('findAll', () => {
+    const route = 'posts';
+    let mock = null;
+
+    describe('Error', () => {
+      it('Should return error when wrong token', async () => {
+        const [status, res] = await getJWTResponse(app, 'get', route);
+
+        expect(res.message).toEqual('Unauthorized');
+        expect(status).toEqual(HttpStatus.UNAUTHORIZED);
+      });
+    });
+
+    describe('Success', () => {
+      beforeEach(() => {
+        mock = mockJwtVerified(user);
+      });
+
+      afterEach(() => {
+        mock.mockRestore();
+      });
+
+      it('Should find all success', async () => {
+        const [status, res] = await getJWTResponse(app, 'get', route);
+
+        expect(res.length).toEqual(posts.length);
+        expect(res).toMatchObject(posts);
+        expect(status).toEqual(HttpStatus.OK);
+      });
+    });
+  });
+
+  describe('findOne', () => {
+    let mock = null;
+    let post: Post;
+    let route: string;
+
+    beforeAll(async () => {
+      post = await create(Post, {
```

Comment:
dùng factory để tạo data test đi anh

Reply:
bản thân hàm create này có dùng useSeederFactory().save() rồi á e. Hay ý e là tạo file seeder nhỉ

Reply:
a tạo data test kiểu này cũng đc, cơ mà e thấy a tạo file post.factory.ts mà ko xài ấy. nếu a ko dùng file đó thì xóa đi nha a, còn nếu dùng thì có thể tạo post bằng file đó

Reply:
@khanhtlb-3057 có dùng các file *.factory.ts e nhé. Ở hàm initApp e check hàm `setFactories` sẽ thấy nó load các file factory ở đây

Reply:
e thấy chỗ đó rồi, chỉ là nếu a tạo post bằng cách này thì nó ghi đè lên các field đã ở post.factory. thành ra file factory trong trường hợp này nó ko có ý nghĩa gì hết ấy a

Reply:
@khanhtlb-3057 nhưng nếu ko tạo file factory cho nó thì khi e dùng `useSeederFactory(entity).save(overrideParams);` nó sẽ báo lỗi `no seeder factory is registered for the entity` nhé.

Reply:
@tanpv-2025 a có thể dùng datasource để tạo cũng đc ấy a

Reply:
@khanhtlb-3057 a mới update lại code factory rồi e nhé, giờ dùng data ở file post.factory.ts rồi :v:

Reply:
ok a